docs/spec/tuples.rst

```diff
@@ -0,0 +1,155 @@
+Tuples
+======
+
+The ``tuple`` class has some special behaviors and properties that make it
+different from other classes from a typing perspective. The most obvious
+difference is that ``tuple`` is variadic -- it supports an arbitrary number
+of type arguments. At runtime, the sequence of objects contained within the
+tuple is fixed at the time of construction. Elements cannot be added, removed,
+reordered, or replaced after construction. These properties affect subtyping
+rules and other behaviors as described below.
+
+
+Tuple Type Form
+---------------
+
+The type of a tuple can be expressed by listing the element types. For
+example, ``tuple[int, int, str]`` is a tuple containing an ``int``, another
+``int``, and a ``str``.
+
+The empty tuple can be annotated as ``tuple[()]``.
+
+Arbitrary-length homogeneous tuples can be expressed using one type and
+ellipsis, for example ``tuple[int, ...]``. This type is equivalent to a union
```

Comment:
Equivalent wrt subtyping or consistent subtyping?

Reply:
I mean "equivalent with respect to subtyping". We may need more precise terminology here. I tend to use "equivalent" to mean that the two types describe _exactly_ the same set of values (from a set theory standpoint), and I use the term "bidirectionally compatible" to describe the concept of two types that are consistent subtypes of each other but not necessarily equivalent. If you have a better suggestion for terminology, let me know.

Reply:
My preference would be to spell out "equivalent wrt ..." explicitly, but having a glossary where "equivalent" and "bidirectionally compatible" are defined would also work.